src/framework/components/element/component.js

```diff
@@ -86,50 +168,109 @@ pc.extend(pc, function () {
                 this._aabbVer++;
             }
 
-            var resx = 0;
-            var resy = 0;
             var screen = this.element.screen;
+            var rect = null;
 
-            if (this.element._anchorDirty) {
-                var px = 0;
-                var py = 1;
-                if (this._parent && this._parent.element) {
-                    // use parent rect
-                    resx = this._parent.element.width;
-                    resy = this._parent.element.height;
-                    px = this._parent.element.pivot.x;
-                    py = this._parent.element.pivot.y;
-                } else if (screen) {
-                    // use screen rect
-                    var resolution = screen.screen.resolution;
-                    resx = resolution.x * screen.screen.scale;
-                    resy = resolution.y * screen.screen.scale;
-                }
-                this.element._anchorTransform.setTranslate((resx*(this.element.anchor.x - px)), -(resy * (py-this.element.anchor.y)), 0);
+            if (this._parent && this._parent.element) {
+                rect = new pc.Vec4(
```

Comment:
This method is basically _sync method of entity – it's called upon transforms getting dirty. But I can definitely pre-alloc that.